blog/2022-10-13-tips-to-secure-kubernetes-cluster/index.md

```diff
@@ -0,0 +1,102 @@
+staging-dev       Active   2d9h
+staging-test      Active   2d9h
+prod-public       Active   2d9h
+```
+
+It’s a good practice to enable namespaces when you want to secure your Kubernetes clusters. Refer to official [Kubernetes namespace documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) to understand how you can use namespaces.
+
+### Enabling Least Privilege Access
+
+Things don’t just stop after enabling RBAC and namespaces. These are just a bare minimum. With large footprints and multiple interfacing points of enterprise applications today, ensuring that each of them is safeguarded is tough. Both these mechanisms allow you to prevent attacks from outside the cluster.
+
+But what if someone portraying a legitimate user manages to enter the cluster? What if there’s an internal threat?
+
+To prevent such attacks, you need to implement least privilege access. The best part is that this is not limited only to users but also applies to applications & accounts.  You enable just enough access to users/accounts/systems that they need.
+
+By enabling least privilege access, you **reduce the attack surface as it limits privileged access exploits**. Malware attacks that require elevated access to move within the system, are also prevented by implementing least privilege access.  
+
+One of the crucial things for implementing least privileged access is to centrally manage access & credentials. Enabling just in time access allows users/systems to have temporary access on need basis only. This further makes implementing and managing least privilege access easier.
+
+### Using 3rd Party Identity Providers
+
+Organizations today use external identity providers to enable access to their enterprise applications for their employees. Identity providers create, maintain and manage identity and provide user authentication as a service. Organisations can configure access to their Kubernetes clusters by using their identity providers like Google, Microsoft, GitHub etc.
+
+Administrators can also create groups with their identity providers and map them to roles and permissions within their Kubernetes clusters. This will enable more robust and end-to-end implementation of RBAC. Further, considering the employee churn, admins only need to modify users at the IdP end as employees move in/out of the organization without modifying roles on their Kubernetes infrastructure.
+
+<img src="/img/docs/paralus-google-login-2.png" alt="Authenticate on Google" height="75%" width="75%"/>
+
+To understand how this would work, you can implement a custom access management solution integrating identity providers with Kubernetes, using open source tools like [Dex and Kubelogin](https://rafay.co/the-kubernetes-current/diy-access-management-using-dex-and-kubelogin/). This is something you can do to understand how things work, but it isn’t something that we recommend as this isn’t a fool proof solution and not recommended for an enterprise setup.
```

Comment:
when we say this isn't something we recommend for enterprise setup ? is there a reasoning behind

Reply:
Mentioned reasons.